### PR TITLE
[DevTools] [Context] Legacy Context

### PR DESCRIPTION
**Pull request by hristo-kanchev**
Fixes: #16466  and #16679

**Description**:
We are now changing the context label in DevTools based on if the component is using the legacy way of adding context.

### CODE REVIEW

**Comment by gaearon** [comment]
Is it possible to add some sort of a test?

**Comment by hristo-kanchev** [comment]
> Is it possible to add some sort of a test?

Tests ✅

**Comment by hristo-kanchev** [comment]
@gaearon Did you manage to check the tests?

File: packages/react-devtools-shared/src/__tests__/inspectedElementContext-test.js [hunk 1]
```diff
@@ -11,10 +11,13 @@ import typeof ReactTestRenderer from 'react-test-renderer';
 import type {GetInspectedElementPath} from 'react-devtools-shared/src/devtools/views/Components/InspectedElementContext';
 import type {FrontendBridge} from 'react-devtools-shared/src/bridge';
 import type Store from 'react-devtools-shared/src/devtools/store';
+import React, {Component, createContext} from 'react';
+import {Fragment} from '../../../../../../../../../../Applications/WebStorm.app/Contents/plugins/JavaScriptLanguage/jsLanguageServicesImpl/flow/react';
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/__tests__/inspectedElementContext-test.js]: 😆

**Comment by bvaughn** [comment]
> @gaearon Did you manage to check the tests?

What's this comment about?

**Comment by gaearon** [comment]
I think this was meant as “@gaearon can you review again now that tests were added”. Lgtm

File: packages/react-devtools-shared/src/__tests__/inspectedElementContext-test.js (hunk 1 above)
Comment (hristo-kanchev) [reply packages/react-devtools-shared/src/__tests__/inspectedElementContext-test.js]: Freakin' WebStorm 😆 #NeverTrustYourEditor

File: packages/react-devtools-shared/src/backend/legacy/renderer.js [hunk 2]
```diff
@@ -753,6 +753,9 @@ export function attach(
       // Can view component source location.
       canViewSource: type === ElementTypeClass || type === ElementTypeFunction,
 
+      // Only legacy context exists in legacy versions.
+      hasLegacyContext: true,
```
Comment (gaearon) [inline packages/react-devtools-shared/src/backend/legacy/renderer.js]: Should we also hide empty one though in the legacy mode?

**Pull request merged**